src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_data_item_schema.cc

```diff
@@ -9,33 +9,78 @@ CREATE_LOGGERPTR_LOCAL(vehicle_data_logger, "VehicleDataItemSchema");
 
 VehicleDataItemSchema::VehicleDataItemSchema(PolicyDataItem& policy_item,
                                              SchemaType schema_type) {
-  policy_table::VehicleDataItem policy_data_item;
+  using namespace policy_table;
+  VehicleDataItem policy_data_item;
   policy_item.getValue(policy_data_item);
 
   auto get_object_schema = [&policy_data_item,
                             &schema_type]() -> decltype(so_schema_item_) {
     smart_objects::Members members;
     auto object_schema = smart_objects::CObjectSchemaItem::create(members);
 
-    for (auto& e : *policy_data_item.params) {
-      auto schema_item = PolicyDataItem(e);
+    std::set<std::string> vehicle_data_names;
+    for (const auto& item : *policy_data_item.params) {
+      vehicle_data_names.insert(item.name);
+    }
+
+    for (const auto& name : vehicle_data_names) {
+      std::vector<VehicleDataItem> full_history_vector;
+      std::copy_if(
+          policy_data_item.params->begin(),
+          policy_data_item.params->end(),
+          std::back_inserter(full_history_vector),
+          [&name](VehicleDataItem& item) { return item.name == name; });
+
+      std::sort(full_history_vector.begin(),
+                full_history_vector.end(),
+                [](const policy_table::VehicleDataItem& left,
+                   const policy_table::VehicleDataItem& right) {
+                  if (!right.since.is_initialized()) {
+                    return false;
+                  }
+                  if (!left.since.is_initialized()) {
+                    return true;
+                  }
+                  const std::string l = *left.since;
+                  const std::string r = *right.since;
+                  return std::stof(l.c_str()) > std::stof(r.c_str());
+                });
+
+      std::vector<smart_objects::SMember> history_vector = {};
```

Comment:
@mkorniichuk std::vector will be initialized an empty by default. Do you need an empty initializer list for vector ?